action/protocol/util/readstate.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2019 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package util
```

Comment:
util is not a good name. it is too general.
can we just put these functionalities under protocol pack

Reply:
I used to until find cyclic dependency: protocol -> rewarding -> protocol